src/cargo/util/config/mod.rs

```diff
@@ -2215,10 +2215,43 @@ impl Drop for PackageCacheLock<'_> {
     }
 }
 
+#[derive(Debug, Default, Deserialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub enum CargoHttpProxyAuth {
+    #[default]
+    Auto,
+    Disable,
+    Basic,
+    Digest,
+    Gss,
+    Ntlm,
+}
+
+impl CargoHttpProxyAuth {
+    pub fn to_easy(&self) -> Auth {
+        let mut easy = Auth::new();
+        let easy = match self {
```

Comment:
Avoid the `easy.clone()` below by not redefining `easy` here. Also consider calling it `auth` instead of `easy`
```suggestion
        match self {
```

Reply:
It's not possible to avoid the clone because `match self { .. }` is a `&mut Auth`. Also `Auth` does not implement `Copy`.

https://docs.rs/curl/0.4.44/curl/easy/struct.Auth.html

Reply:
You don't need to capture the result of the `match` in a variable. The `basic()`, `digest()`, etc methods mutate the `Auth` struct, so you can return the one you created with `Auth::new()` without cloning.
```
    pub fn to_easy(&self) -> Auth {
        let mut auth = Auth::new();
        match self {
            Self::Auto => auth.basic(true).digest(true).gssnegotiate(true).ntlm(true),
            Self::Disable => &auth,
            Self::Basic => auth.basic(true),
            Self::Digest => auth.digest(true),
            Self::Gss => auth.gssnegotiate(true),
            Self::Ntlm => auth.ntlm(true),
        };
        auth
    }
```